src/bin/pg_autoctl/file_utils.c

```diff
@@ -127,12 +156,10 @@ ensure_empty_dir(const char *dirname, int mode)
 bool
 write_file(char *data, long fileSize, const char *filePath)
 {
-	FILE *fileStream = NULL;
 
-	fileStream = fopen(filePath, "wb");
+	FILE *fileStream = fopen_with_umask(filePath, "wb", O_WRONLY | O_CREAT, 0644);
 	if (fileStream == NULL)
 	{
-		log_error("Failed to open file \"%s\": %s", filePath, strerror(errno));
```

Comment:
Please add a comment about not logging on purpose, as we do everywhere in the code:

``` c
    /* errors have already been logged */
```